### PR TITLE
[New] Add json2csv w/ npm auto-update via single package.json

### PR DESCRIPTION
**Pull request by waseemahmad31**
# Pull request template for new library

Pull request for issue: #13197 

## Profile of the lib

* Git repository (required): https://github.com/zemirco/json2csv
* Official website (optional, not the repository): http://www.mircozeiss.com/json2csv/
* NPM package url (optional): https://www.npmjs.com/package/json2csv
* License and its reference: MIT
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 31
  * Count of stars: 1526
  * Count of forks: 260

## Essential checklist

* [ ] I'm the author of this library
  * [ ] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [x] Auto-update target/source is valid.
* [x] Auto-update filemap is correct.

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [ ] Separate unrelated changes into different commits.
* [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [ ] Close corresponding issue in commit message
* [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/json2csv.cjs/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "name": "Mirco Zeiss",
+    "email": "mirco.zeiss@gmail.com",
+    "twitter": "zeMirco"
+  },
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/zemirco/json2csv"
+  },
+  "autoupdate": {
```
Comment (extend1994) [inline ajax/libs/json2csv.cjs/package.json]: Please use npm auto-update to fetch `dist`

File: ajax/libs/json2csv.cjs/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "json2csv.cjs",
+  "filename": "json2csv.cjs.js",
```
Comment (extend1994) [inline ajax/libs/json2csv.cjs/package.json]: UMD file is the one for a browser, please use `json2csv.umd.min.js`, we will minify it :)

File: ajax/libs/json2csv.cjs/package.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "json2csv.cjs",
```
Comment (extend1994) [inline ajax/libs/json2csv.cjs/package.json]: Please use npm package name.

File: ajax/libs/json2csv.cjs/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "url": "https://github.com/zemirco/json2csv"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/zemirco/json2csv.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*"
```
Comment (extend1994) [inline ajax/libs/json2csv.cjs/package.json]: Please use `**/!(*.esm.*|*.cjs.*)` which cdnjs doesn't need.

**Comment by waseemahmad31** [comment]
Thanks @extend1994 , for review comments. I will change it as per the suggesion.

**Comment by waseemahmad31** [comment]
I have renamed the folder `json2csv.cjs` to `json2csv`. But don't know why **CI** is trying to get the folder `json2csv.cjs` and its `package.json` in `sparseCheckout config`. Getting the error `/ajax/libs/json2csv.cjs/package.json not found!!!`. Please help me with this.

**Comment by PeterDaveHello** [comment]
@waseemahmad31 because the filename is still in the commit "history", I'll help you squash these commits :)

**Comment by waseemahmad31** [comment]
> @waseemahmad31 because the filename is still in the commit "history", I'll help you squash these commits :)

thanks @PeterDaveHello

**Comment by PeterDaveHello** [comment]
@waseemahmad31 done, take a look :)

**Comment by waseemahmad31** [comment]
Great 👍 .

**Comment by waseemahmad31** [comment]
This pull request is pending from long time. Anything required from my side to close this pull request?

**Pull request merged**